enclave/core/sgx/init.c

```diff
@@ -44,30 +44,16 @@ static void _check_memory_boundaries(void)
 extern volatile const oe_sgx_enclave_properties_t oe_enclave_properties_sgx;
 extern oe_eeid_t* oe_eeid;
 
-static int _is_eeid_base_image(
-    const volatile oe_sgx_enclave_properties_t* properties)
+static void _find_eeid()
 {
-    return properties->header.size_settings.num_heap_pages == 0 &&
-           properties->header.size_settings.num_stack_pages == 0 &&
-           properties->header.size_settings.num_tcs == 1;
-}
-
-static oe_result_t _eeid_patch_memory()
-{
-    oe_result_t r = OE_OK;
-
-    if (_is_eeid_base_image(&oe_enclave_properties_sgx))
-    {
-        uint8_t* enclave_base = (uint8_t*)__oe_get_enclave_base();
-        uint8_t* heap_base = (uint8_t*)__oe_get_heap_base();
-        oe_eeid_marker_t* marker = (oe_eeid_marker_t*)heap_base;
-        oe_eeid = (oe_eeid_t*)(enclave_base + marker->offset);
-
-        // Wipe the marker page
-        memset(heap_base, 0, OE_PAGE_SIZE);
-    }
-
-    return r;
+    /* Get the last page before the heap. */
+    uint8_t* heap_base = (uint8_t*)__oe_get_heap_base();
```

Comment:
It would be better to update the heap pointers to exclude the eeid page. Otherwise, the last page might get allocated via malloc, wiping out the eeid.

Reply:
Oh yes, that's there but hard to see! There is an extra page just before the heap starts, that's why in the next line I take `heap_base - OE_PAGE_SIZE`. That's a bit of a hack though, I should look for a cleaner solution to this, e.g. patching an rva into the image.

Reply:
(Done0